java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1312,50 +1349,77 @@ public final ColumnBuilder appendNull() {
       setNullAt(currentIndex);
       currentIndex++;
       if (type == DType.STRING) {
-        initAndResizeOffsetBuffer();
+        initAndResizeOffsetBuffer(currentIndex);
         offsets.setInt(currentIndex * OFFSET_SIZE, currentByteIndex);
       }
       return this;
     }
 
-    private ColumnBuilder append(List inputList) {
-      assert type == DType.LIST;
-      // We know lists have only 1 child
-      ColumnBuilder childBuilder = childBuilders.get(0);
-      if (inputList == null) {
-        initValidBuffer();
-        setNullAt(currentIndex);
+    // List<StructData> when struct
+    private <T> ColumnBuilder append(List<T> inputList) {
+      assert type.isNestedType();
+      if (type == DType.STRUCT) {
+        assert !inputList.isEmpty();
+        //TODO: assumption
+        int numChildren = inputList.size();
+        for (int i = 0; i < numChildren; i++) {
+          StructData structData = (StructData) inputList.get(i);
+          if (structData.dataRecord == null) {
+            setNullAt(currentIndex);
```

Comment:
One of the things we have to be sure we do is that when we set a null at the parent struct level we reflect that null in the children as well.  This is one of the assumptions that we made in the cudf side to make pulling data out of a struct cheap.

Reply:
I will add a test to verify this.

Reply:
@revans2  The latest changes do not address this one comment. I will cover it now in the next set of changes.